### PR TITLE
fabric: Add new protocol type and address format for PSM2

### PR DESCRIPTION
**Pull request by j-xiong**
The protocols and address formats of the psm and psm2 providers are
different and should not share the same definition.

Signed-off-by: Jianxin Xiong <jianxin.xiong@intel.com>

Address #3047

### CODE REVIEW

File: src/common.c [hunk 1]
```diff
@@ -337,6 +343,19 @@ static int ofi_str_to_psmx(const char *str, void **addr, size_t *len)
...
+{
+	int ret;
+
+	*len = 2 * sizeof(uint64_t);
+	if (!(*addr = calloc(1, *len)))
+		return -FI_ENOMEM;
+
+	ret = sscanf(str, "%*[^:]://%" SCNu64 ":%" SCNu64,
+		     (uint64_t *) *addr, (uint64_t *) *addr + 1);
+	return (ret == 2) ? 0 : -FI_EINVAL;
```
Comment (shefty) [inline src/common.c]: We should probably free addr if we return a failure.  I'm guessing I have this error in the other str_to_xxx calls as well.

Comment (j-xiong) [inline src/common.c]: Updated. The error existed only in the two psm related functions.

**Comment by shefty** [comment]
Thanks - I was just logging into my system to create patches to fix this.

**Comment by j-xiong** [comment]
Updated again to use the same format for sscanf and snprintf otherwise one can't get back the original result with two conversions.

**Pull request merged**